tests/shutdown.rs

```diff
@@ -579,3 +580,55 @@ fn timely_reload_shutdown() {
         );
     });
 }
+
+#[tokio::test]
+async fn health_503_during_shutdown() {
+    use std::process::Command;
+
+    let mut cmd = Command::cargo_bin("vector").unwrap();
+
+    cmd.arg("--quiet")
+        .arg("-c")
+        .arg(create_file(
+            r#"
+            [api]
+              enabled = true
+              address = "127.0.0.1:8686"
+                    
+            [sources.source]
+              type = "demo_logs"
+              format = "json"
+              interval = 0
+            
+            [sinks.sink]
+              type = "blackhole"
+              inputs = ["source"]
+              rate = 1
+            "#,
+        ))
+        .env("VECTOR_DATA_DIR", create_directory());
+
+    let mut vector = cmd
+        .stdin(std::process::Stdio::piped())
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    // Give vector time to start.
+    sleep(STARTUP_TIME);
```

Comment:
I know this was here already, but this feels like a potential source for flaky tests. We could use the health endpoint to check for `{ok: true}` on a loop with a timeout, and maybe shave some secs off each integration test.